### ISSUE TITLE
Only show valid filter combinations in Filter Box 

### ISSUE THREAD

**Issue by durchgedreht** [apache/superset#10688]
Hi,
is it possible to have a filterbox with mutiple filters whose filter options are dependant on the already selected entries?

Let's say we have the following table "sold_products":
```
customer  | product | product_category | revenue
----------+---------+------------------+-----------
alice     | teddy   | toys             | 20
alice     | doll    | toys             | 10
alice     | skirt   | clothes          | 25
bob       | car     | toys             | 15
bob       | holmes  | books            | 10
``` 

Then I create a Filter Box with 

- customer
- product
- product_category

I would like to achieve the following:

- When I select "customer" filter = "alice" products filter just shows ["teddy", "doll", "skirt"]
- When I select "product_category" filter ="toys" products filter just shows ["teddy", "doll", "car"]
- When I select "cutomser" filter ="alice" AND "product_category" filter ="toys" products filter just shows ["teddy", "doll"]

Currently all values are always shown. Is there a way to achieve the behaviour described?  
I know this will cause some  DB stress, which is acceptable. Thanks!

**Comment by JunlinC** [comment]
Cascading filter feature is currently in our roadmap. we will announce it when it's available. :)

**Comment by Francoislinkt** [comment]
The only way to have it working I've found is to use 1 filter chart per cols.

Done this way I've got a kind of "contextual" filter.

**Comment by Zverik** [comment]
Duplicate of #10013. Implemented in #5664 two years ago, but not merged because in a year or two we'll get dashboard filters (#10285).

**Comment by junlincc** [comment]
> Duplicate of #10013. Implemented in #5664 two years ago, but not merged because in a year or two we'll get dashboard filters (#10285).
> 
> Also, one filter chart per column does not work.
@Zverik @durchgedreht @Francoislinkt 
we looked at the old cascading filter PR, if we had merged it, it would introduce quite some complexity. 
the good news is, dashboard native cascading filter(not filter box) is on its way, you should expect it to be fully available in a few weeks.
